python/lsst/ts/scriptqueue/ui/request_model.py

```diff
@@ -0,0 +1,547 @@
+            state['current'] = self.get_script_info(self.state['current'])
+
+        return state
+
+    def get_script_info(self, salindex):
+        """Get info about a script with `salindex`.
+
+        Parameters
+        ----------
+        salindex : int
+            Index of the sal script to get info.
+
+        Returns
+        -------
+        script_info : dict
+
+        """
+        if (salindex in self.scripts and
+                self.scripts[salindex]['process_state'] >= ScriptProcessState.DONE):
+            # if a script is in a final state it will not be updated anymore so
+            # just retrieve stored information
+            # self.log.debug(f"Done {salindex}")
+            self.scripts[salindex]['updated'] = False
+            return self.scripts[salindex]
+        elif salindex in self.scripts:
+            # if script is in the list update all script info and grab the stored information
+            # if the script was updated in updata_script_info, mark it as such, otherwise
+            # flag it as not updated.
+            # self.log.debug(f"Updating {salindex}")
+            updated = self.update_script_info()
+            self.scripts[salindex]['updated'] = salindex in updated
+            return self.scripts[salindex]
+        else:
+            # It may happen that a script is in the queue but no information is available,
+            # for instance, while it is still pre-loading. In this case, pass an empty
+            # state.
+            # self.log.debug(f"Adding {salindex}")
+            info = self.queue.evt_script.DataType()
+            info.salIndex = salindex
+            # this will add the script to the list of script, but there's no real information about it
+            # except for the sal index
+            self.parse_script_info(info)
+            self.scripts[salindex]['updated'] = True
+            return self.scripts[salindex]
+
+    def add(self, path, is_standard, config):
+        """Add a script to the queue.
+
+        Parameters
+        ----------
+        path : str
+            Name of the script.
+        is_standard : bool
+            Is it a standard script?
+        config : str
+            YAML configuration string.
+
+        Returns
+        -------
+        salindex : int
+            The salindex of the script added to the queue.
+        """
+
+        self.queue.cmd_add.set(isStandard=is_standard,
+                               path=path,
+                               config=config,
+                               location=SALPY_ScriptQueue.add_Last)
+
+        ack = self.run(self.queue.cmd_add.start(timeout=self.cmd_timeout))
+
+        return int(ack.ack.result)
+
+    def set_queue_log_level(self, level):
+        """Set queue log level.
+
+        Parameters
+        ----------
+        level : int
+            Log level; error=40, warning=30, info=20, debug=10
+
+        """
+        valid_level = [40, 30, 20, 10]
```

Comment:
Why this check? The logging module allows any positive (?) integer as a level (though some levels have names -- i.e. the ones you list here).